src/KafkaFlow.Serializer.SchemaRegistry.ConfluentAvro/ConsumerConfigurationBuilderExtensions.cs

```diff
@@ -17,9 +17,10 @@ public static class ConsumerConfigurationBuilderExtensions
         public static IConsumerMiddlewareConfigurationBuilder AddSchemaRegistryAvroSerializer(
             this IConsumerMiddlewareConfigurationBuilder middlewares)
         {
-            return middlewares.AddSerializer(
-                resolver => new ConfluentAvroSerializer(resolver),
-                resolver => new SchemaRegistryTypeResolver(new ConfluentAvroTypeNameResolver(resolver.Resolve<ISchemaRegistryClient>())));
+            return middlewares.Add(
```

Comment:
Why there are no new extension methods to support async versions on https://github.com/Farfetch/kafkaflow/blob/master/src/KafkaFlow.Serializer/ConsumerMiddlewareConfigurationBuilderExtensions.cs ?

Reply:
@massada @filipeesch

Reply:
Hi @alek-talabat, there were no changes to the public contracts, what do you mean by a breaking change?

Reply:
@massada  - As `SchemaRegistryTypeResolver` implements `IAsyncMessageTypeResolver` now, it is not possible to use `AddSerializer` (from [here](https://github.com/Farfetch/kafkaflow/blob/master/src/KafkaFlow.Serializer/ConsumerMiddlewareConfigurationBuilderExtensions.cs ))anymore, as it works only with `IMessageTypeResolver`. So after updating the package which was using it, it does not compile. And it's not that straightforward to find how to migrate to the syntax as provided in here.

Reply:
Yes, I'm using a custom resolver. I thought of not modifying the existing `AddSerializer` methods, but simply adding new ones which would accept generic parameter of type `IAsyncMessageTypeResolver `. I think that should resolve it

Reply:
This change/PR broke the custom type resolvers feature completely. The overloads are there but are not getting called correctly. See #305